File: b-go/ac/T795.go
package ac

import "fmt"

// 前缀和 https://www.acwing.com/problem/content/797/
// S[i] = S[i - 1] + a[i]，i要从1开始算
// a[l] + ... + a[r] = S[r] - S[l - 1]
func T795() {
    var n, t, x, l, r int
    fmt.Scan(&n, &t)
    q := make([]int, n + 1)
    for i := 1; i <= n; i++ {
        fmt.Scan(&x)
        q[i] = q[i - 1] + x
    }
    for i := 1; i <= t; i++ {
        fmt.Scan(&l, &r)
        fmt.Println(q[r] - q[l - 1])
    }
}

File: a-java/ac/T795.java
package ac;

import java.util.*;

/**
 * 前缀和 https://www.acwing.com/problem/content/797/
 * S[i] = S[i - 1] + a[i]，i要从1开始算
 * a[l] + ... + a[r] = S[r] - S[l - 1]
 */
class T795 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int t = sc.nextInt();
        int[] s = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            int a = sc.nextInt();
            s[i] = s[i - 1] + a;
        }
        
        for (int i = 0; i < t; i++) {
            int l = sc.nextInt();
            int r = sc.nextInt();
            System.out.println(s[r] - s[l - 1]);
        }
        sc.close();
    }
}
